src/_pytest/runner.py

```diff
@@ -467,29 +466,33 @@ class SetupState:
     """
 
     def __init__(self) -> None:
-        # Maps node -> the node's finalizers.
         # The stack is in the dict insertion order.
-        self.stack: Dict[Node, List[Callable[[], object]]] = {}
-
-    _prepare_exc_key = StoreKey[Union[OutcomeException, Exception]]()
+        self.stack: Dict[
+            Node,
+            Tuple[
```

Comment:
Wouldn't this be a case for typing.NamedTuple

Reply:
I personally prefer it inline, that way everything is localized inside the class which makes it easier to follow for me. But I can definitely change it (probably to an attrs class) if you think it'd be better - let me know.

Reply:
I agree with your assessment as we only use it in one place